### PR TITLE
Initial Rust GitHub Action

### PR DESCRIPTION
**Pull request by radu-matei**

### CODE REVIEW

**Review by technosophos — APPROVED** [review]
:shipit:

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Build
+      run: cargo build --verbose
+    - name: Run tests
+      run: cargo test --verbose
```
Comment (glyn) [inline .github/workflows/rust.yml:15]: I'm not convinced that `--verbose` is helpful in most cases. How often do we need to look through _successful_ tests. It's much more usual that we want to see any failing tests in a CI run.
```suggestion
      run: cargo test
```

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: Rust
```
Comment (glyn) [inline .github/workflows/rust.yml:1]: Does simply naming the action `Rust` ensure that rust/cargo are installed?

Comment (technosophos) [reply .github/workflows/rust.yml:1]: Rust is included in the base Ubuntu image in GitHub Actions.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (technosophos) [reply .github/workflows/rust.yml:15]: Yeah, I think I would prefer turning off Verbose and turning on BACKTRACE (can't remember the exact name of that var).

**Comment by technosophos** [comment]
@radu-matei you wanna amend the commit to pass DCO? Then I will merge.

**Comment by radu-matei** [comment]
Sure - I completely forgot about this PR..

Amended and rebased.

**Pull request merged**